src/main/java/org/zalando/nakadi/service/subscription/zk/CuratorZkSubscriptionClient.java

```diff
@@ -290,4 +291,23 @@ public void transfer(final String sessionId, final Collection<Partition.Partitio
             throw new NakadiRuntimeException(e);
         }
     }
+
+    @Override
+    public ZkSubscriptionNode getZkSubscriptionNode() {
+        final ZkSubscriptionNode subscriptionNode = new ZkSubscriptionNode();
+        try {
+            runLocked(() -> {
+                subscriptionNode.setPartitions(listPartitions());
+                subscriptionNode.setSessions(listSessions());
+            });
+        } catch (final NakadiRuntimeException nre) {
+            final Exception cause = nre.getException();
+            if (!(cause instanceof KeeperException.NoNodeException)) {
+                throw new NakadiRuntimeException(cause);
+            }
+            log.info("No data about provided subscription {} in ZK", subscriptionId);
```

Comment:
I think for user it sounds like "event type doesn't have partitions"
I think it's better just to return some specific status code in this situation and an empty body